src/main/java/Duke.java

```diff
@@ -1,10 +1,228 @@
+                    if (description.length() == 0) {
+                        throw new DukeException("☹ OOPS!!! The description of a deadline cannot be empty.");
+                    }
+                    String byID = parts[1].trim();
+                    Task currTask = new Deadline(description, byID);
+                    tasks.add(currTask);
+//                    writeLine(currTask.toTxtString());
+                } catch (StringIndexOutOfBoundsException e) {
+                    throw new DukeException("☹ OOPS!!! The description of a deadline cannot be empty.");
+                }
+            } else if (command.startsWith("event")) {
+                try {
+                    String[] parts = command.split("/from");
+                    String description = parts[0].substring(6).trim();
+                    if (description.length() == 0) {
+                        throw new DukeException("☹ OOPS!!! The description of an event cannot be empty.");
+                    }
+                    String[] timeParts = parts[1].split("/to");
+                    String start = timeParts[0].trim();
+                    String end = timeParts[1].trim();
+                    Task currTask = new Event(description, start, end);
+                    tasks.add(currTask);
+//                    writeLine(currTask.toTxtString());
+                } catch (StringIndexOutOfBoundsException e) {
+                    throw new DukeException("☹ OOPS!!! The description of an event cannot be empty.");
+                }
+            } else if (command.startsWith("delete")) {
+                try {
+                    if (separateCommand.length > 2 || Integer.parseInt(separateCommand[1]) > tasks.size()) {
+                        throw new DukeException("☹ OOPS!!! Invalid number");
+                    }
+                    int taskNumber = Integer.parseInt(separateCommand[1]);
+                    if (command.startsWith("delete")) {
+                        System.out.println(" Noted. I've removed this task:");
+                        System.out.println("   " + tasks.get(taskNumber - 1).toString());
+                        tasks.remove(taskNumber - 1);
+                    }
+                } catch (NumberFormatException | ArrayIndexOutOfBoundsException e) {
+                    throw new DukeException("☹ OOPS!!! Invalid number");
+                }
+            } else {
+                throw new DukeException("☹ OOPS!!! I'm sorry, but I don't know what that means :-(");
+            }
+            if (command.startsWith("delete")) {
+                System.out.println(" Now you have " + tasks.size() + " tasks in the list.");
+            } else {
+                System.out.println(" I've added this task:" + "\n" + "   " + tasks.get(tasks.size() - 1).toString()
+                        + "\n" + " Now you have " + tasks.size() + " tasks in the list.");
+            }
+            writeLine(filePath);
+        } else {
```

Comment:
these if-else statements can be greatly improved to a clearer presentation after you do A-MoreOOP. Do not nest too many things inside a method.